spanner/api/Spanner.Samples/UpdateDataWithTimestampColumnAsync.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_update_data_with_timestamp_column]
```

Comment:
@skuruppu Should this sample use UPDATE...WHERE? This is a very weird way of updating rows, i.e. while they are being read?

Reply:
That's a very good point. You're right and none of the other languages do it this way. They use the update API.

@laljikanjareeya can you take a look at the [other samples](https://cloud.google.com/spanner/docs/commit-timestamp#update-row) and make this one consistent?

Reply:
@amanda-tarafa I did try with below code snippet but it throws Error any Idea:
```cs
SpannerCommand updateCmd = connection.CreateDmlCommand(
    "UPDATE Albums SET LastUpdateTime = @CommitTimestamp "
    + "WHERE SingerId = 2 and AlbumId = 2");
cmd.Parameters.Add("CommitTimestamp", SpannerDbType.Timestamp, SpannerParameter.CommitTimestamp);
await updateCmd.ExecuteNonQueryAsync();
```
Error:
 `
Status(StatusCode="InvalidArgument", Detail="Request contains an invalid argument.", DebugException="Grpc.Core.Internal.CoreErrorDetailException: {"created":"@1601872758.999000000","description":"Error received from peer ipv4:142.250.67.170:443","file":"T:\src\github\grpc\workspace_csharp_ext_windows_x64\src\core\lib\surface\call.cc","file_line":1062,"grpc_message":"Request contains an invalid argument.","grpc_status":3}")
`

Reply:
Ah ok so the issue is that you're not meant to specify an explicit commit timestamp here. It's meant to be the placeholder string "spanner.commit_timestamp()" so that the backend can evaluate that to the timestamps at which this update is being committed. Is it not possible to set this in C# because of some restriction with types? I haven't tried this before in C# but the other languages have a constant to represent this string.

Reply:
My bad, I now realize that this is what you're using.

https://github.com/googleapis/google-cloud-dotnet/blob/8eda274f0fe6e7fa4301da9d44b2662e25771646/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/CommitTimestamp.cs#L29

I'm not sure why this error comes up then. You may have to try to get a more detailed error message from the gRPC layer. I think we've been finding that in other languages, gRPC is swallowing up error message details so the client library is not showing all the relevant error info. Not sure if that's also happening here.

Reply:
Done using `CreateUpdateCommand` it works fine.